src/Rarity/Rarity.router.ts

```diff
@@ -1,18 +1,97 @@
 import { server } from 'decentraland-server'
-
+import { Request } from 'express'
 import { Router } from '../common/Router'
 import { collectionAPI } from '../ethereum/api/collection'
+import { RarityFragment } from '../ethereum/api/fragments'
+import { HTTPError, STATUS_CODES } from '../common/HTTPError'
+import { isUsingRaritiesWithOracle, getRarityFromBlockchain } from './utils'
 
 export class RarityRouter extends Router {
   mount() {
-    /**
-     * Returns the available rarities
-     */
+    // Returns the available rarities.
     this.router.get('/rarities', server.handleRequest(this.getRarities))
+
+    if (isUsingRaritiesWithOracle()) {
+      // Returns a single rarity according to the rarity name provided.
+      this.router.get('/rarities/:name', server.handleRequest(this.getRarity))
+    }
   }
 
-  async getRarities() {
+  getRarities = async (req: Request) => {
+    // If the server is still using the old rarities contract.
+    // Return rarities as they have been always returned
+    if (!isUsingRaritiesWithOracle()) {
+      return collectionAPI.fetchRarities()
+    }
+
+    const inUSD = req.query.inUSD === 'true'
```

Comment:
What if we name this query parameter `currency`? We can later have different conversions if we want to.
```suggestion
    const inUSD = req.query.currency === 'usd'
```

Reply:
I like that!